packages/astro/src/@types/astro.ts

```diff
@@ -1972,6 +1972,11 @@ interface AstroSharedContext<
 	 * Object accessed via Astro middleware
 	 */
 	locals: App.Locals;
+
+	/**
+	 * The current locale that is computed from the `Accept-Language` header of the browser (**SSR Only**).
+	 */
```

Comment:
I don't think it's possible because `Headers` aren't available in SSG:

> [ssg] Headers are not exposed in static (SSG) output mode. To enable headers: set `output: "server"` in your config file.

Do you think we should have a fallback? And if so, how? By reading the current URL of `Request`?

Reply:
Yep! This could be derived from the URL in SSG, if it matches a known `/[lang]/...` route. Suggested that here https://github.com/withastro/astro/pull/8935#discussion_r1374678110

Reply:
We don't send an Accept-Language header during an SSG build though, right? I'm not sure, how would we add such functionality?

Reply:
The use-case here is, for example, the user is navigated to `/es` but their Accept-Language is `en` and you might display a notice that they are viewing a different language. If we had a helper that returns the current locale from the URL I would expect that to be a different helper (I don't think the RFC currently defines one).

Reply:
Yeah, I believe the value coming from the URL and the value coming from the header are different, semantically. What do you think if I rename `currentLanguage` to `preferredLanguage`? Because that's the actual meaning. cc @matthewp @natemoo-re 

A property called `currenLanguage` would actually read the URL of the request.

Reply:
Yeah, good call on the rename!

> A property called `currenLanguage` would actually read the URL of the request.

Unless we have a use-case for this, I'm not sure that we should add it

Reply:
Yep, thanks for clarifying. `preferredLanguage` seems fine to me and I withdraw my suggestion for SSG support!

Reply:
I think Chris might be a good person to run this design by, though, given that Starlight only supports SSG but does manage translations with fallback content.